src/clp_ffi_py/ir/native/deserialization_methods.cpp

```diff
@@ -262,16 +262,15 @@ auto deserialize_preamble(PyObject* Py_UNUSED(self), PyObject* py_deserializer_b
                 static_cast<char const*>(clp::ffi::ir_stream::cProtocol::Metadata::VersionKey)
         )};
         auto const error_code{clp::ffi::ir_stream::validate_protocol_version(version)};
-        if (IRProtocolErrorCode::IRProtocolErrorCode_Supported != error_code) {
+        if (IRProtocolErrorCode::BackwardCompatible != error_code) {
```

Comment:
```suggestion
        // TODO: Check against `Supported` instead of `BackwardCompatible` once support for Key-value Pair IR Stream is
        // added in `clp-ffi-py`.
        if (IRProtocolErrorCode::BackwardCompatible != error_code) {
```

Reply:
I added a comment below explaining why `Supported` is actually not supported by this method.
This method should eventually be removed after the kv pair IR deserializer has backward-compatible support, so I guess it's ok not to explain why using `BackwardCompatible` here?

Reply:
sure